src/UKHO.S100PermitService.API/Program.cs

```diff
@@ -131,9 +133,19 @@ private static void ConfigureServices(WebApplicationBuilder builder)
             });
 
             builder.Services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+
             builder.Services.AddScoped<IPermitService, PermitService>();
             builder.Services.AddScoped<IFileSystem, FileSystem>();
             builder.Services.AddScoped<IPermitReaderWriter, PermitReaderWriter>();
+
+            builder.Services.AddHttpClient<IHoldingsApiClient, HoldingsApiClient>(client =>
+            {
+                client.BaseAddress = new Uri(configuration.GetValue<string>("HoldingsServiceApiConfiguration:BaseUrl"));
+            });
+            builder.Services.AddDistributedMemoryCache();
+            builder.Services.AddSingleton<IAuthHoldingsServiceTokenProvider, AuthTokenProvider>();
+            builder.Services.AddScoped<IHoldingsService, HoldingsService>();
+            builder.Services.AddTransient<IHoldingsApiClient, HoldingsApiClient>();
```

Comment:
organize the services

example:

builder.Services.AddHttpClient<IHoldingsApiClient, HoldingsApiClient>(client =>
            {
                client.BaseAddress = new Uri(configuration.GetValue<string>("HoldingsServiceApiConfiguration:BaseUrl"));
            });

builder.Services.AddDistributedMemoryCache();

builder.Services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
builder.Services.AddSingleton<IAuthHoldingsServiceTokenProvider, AuthTokenProvider>();

builder.Services.AddScoped<IPermitService, PermitService>();
builder.Services.AddScoped<IFileSystem, FileSystem>();
builder.Services.AddScoped<IPermitReaderWriter, PermitReaderWriter>();
builder.Services.AddScoped<IHoldingsService, HoldingsService>();

builder.Services.AddTransient<IHoldingsApiClient, HoldingsApiClient>();

Reply:
Resolved